### ISSUE TITLE
Link of an image

### ISSUE THREAD

**Issue by BrightkyEfoo** [WakHerv7/Test_API_Ecommerce_Agence_Digitale#1]
How to get the link of an image after creating a product?
I get the name but I don't know wich link or url should i use to render it in a html image element

**Comment by WakHerv7** [comment]
Sorry, I should have mentioned this:

I saved images in differents folders:

To get product thumbnails:
/uploads/thumbnailsProduits/[filename]
Example: http://localhost:3333/uploads/thumbnailsProduits/clv8fqfke00058cjjbxggh0lr.png

To get product image:
/uploads/imagesProduits/[filename]
Example: http://localhost:3333/uploads/imagesProduits/clv8fqfku00078cjjfv190ysh.png

To get category image:
/uploads/categories/[filename]
Example: http://localhost:3333/uploads/categories/clv8efig40001v5jj4mqx56wa.jpg

Actually, I should've added the file path instead of the file name.

**Comment by BrightkyEfoo** [comment]
Yes providing the file path is better for the API consumer. It make sense to have directly a link, 
'Cause at any time for any reason you can be call to change the location of your API, I mean a differnt ip, a different port or just a different domain name.

**Comment by BrightkyEfoo** [comment]
But I notice that every user can add and remove any product, Is it a good way ?

**Comment by WakHerv7** [comment]
I didn't realize it back then, since I tested it with only one user.
No it's not a good way. 
It depends :
- If the user is a super admin, then he can add and remove any product.
- If the user is a shop owner, then should only add or remove products in his shop.
- If the user is a customer, then he should not add or remove any product at all.

**Comment by BrightkyEfoo** [comment]
It means you have to introduce lot of new thinks, I mean this is a kind of RBAC

**Comment by WakHerv7** [comment]
Yes it is.
But the test didn't mention anything like that. That's why I didn't do it.

**Comment by BrightkyEfoo** [comment]
Hummm, ok ok. it will be nice If we can work one day on a project together, Pleas' can i have your email address ?

**Comment by WakHerv7** [comment]
wakam.hermann@gmail.com

Are you the examinator or what ?

**Comment by BrightkyEfoo** [comment]
I'm a competitor

**Comment by BrightkyEfoo** [comment]
Yo didn't notice?

**Comment by WakHerv7** [comment]
Why are you asking me all those questions ?
Do you want to understand my work to better your own ? It's not fair.

**Comment by BrightkyEfoo** [comment]
![image](https://github.com/WakHerv7/Test_API_Ecommerce_Agence_Digitale/assets/108243838/bc585c70-251a-4546-b511-45a0592324d8)
I already submit my work since two days bro, don't worry. I'm not here to cheat on you.

**Comment by BrightkyEfoo** [comment]
Don't worry, chill chill

**Comment by WakHerv7** [comment]
Okay. Did you use Adonis too ?

**Comment by BrightkyEfoo** [comment]
> Why are you asking me all those questions ? Do you want to understand my work to better your own ? It's not fair.

I'm asking you all those, like, I want understand. I'm in love with your code, I love when things they good. Nothing more!

**Comment by BrightkyEfoo** [comment]
> Okay. Did you use Adonis too ?

Nope, actually I'm learning adonisJS. You see why I'm so interested by your code, right ?

**Comment by WakHerv7** [comment]
I see you did backend and frontend.

I learned AdonisJs in 02 days, to be able to do the test. Initially i applied for frontend, but the test was for backend so I adapted. I already know some backend (Laravel, Django). AdonisJs is very similar to Laravel. That's why I did it.

I thought the deadline was last sunday. That's why I didn't do any frontend.

**Comment by BrightkyEfoo** [comment]
Yes I did them two. 
No the deadline is this sunday. Is it really possible to do it in 3 days ?

**Comment by WakHerv7** [comment]
It's possible with AdonisJS.

**Comment by BrightkyEfoo** [comment]
Humm, Humm, you did it

**Comment by BrightkyEfoo** [comment]
Ok see you, I have a lot of work now... I will mail you later

**Comment by WakHerv7** [comment]
Ok, me too.
